src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+            "Unable to parse the container name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not blobName:
+        raise ValueError(
+            "Unable to parse the blob name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not sasToken:
+        raise ValueError(
+            "Unable to parse the SAS token from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    blobService = AppendBlobService(account_name=accountName,
+                                    sas_token=sasToken)
+
+    byteSize = 1024*4
+    timeoutInMinutes = 10
+    timeoutInSeconds = timeoutInMinutes * 60
+    now = time.time()
+    stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName)
+
+
+def get_match(sasURL):
+    return re.search(
+        (r"http(s)?://(?P<accountName>.*?)\..*?/(?P<containerName>.*?)/"
+            r"(?P<blobName>.*?)\?(?P<sasToken>.*)"), sasURL)
+
+
+def stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName):
+
+    colorama.init()
+    stream = BytesIO()
+    metadata = dict()
+    start = 0
+    end = byteSize - 1
+    available = 0
+    lastModified = None
+
+    # Try to get the initial properties so there's no waiting.
+    # If the storage call fails, we'll just sleep and try again after.
+    try:
+        props = blobService.get_blob_properties(
+            container_name=containerName, blob_name=blobName)
+        metadata = props.metadata
+        available = props.properties.content_length
+        lastModified = props.properties.last_modified
+    except:
+        pass
+
+    while (blob_is_not_complete(metadata) or start < available):
+
+        while start < available:
+            try:
+                oldByteSize = len(stream.getvalue())
+                blobService.get_blob_to_stream(
+                    container_name=containerName,
+                    blob_name=blobName,
+                    start_range=start,
+                    end_range=end,
+                    stream=stream)
+
+                currBytes = stream.getvalue()
+                newByteSize = len(currBytes)
+                amountRead = newByteSize - oldByteSize
+                start += amountRead
+                end = start + byteSize - 1
+
+                # Only scan what's newly read. If nothing is read,
+                # default to 0.
+                minScanRange = max(newByteSize - amountRead - 1, 0)
+                for i in range(newByteSize - 1, minScanRange, -1):
+                    if currBytes[i-1:i+1] == b'\r\n':
+                        flush = currBytes[:i]  # won't print \n
+                        stream = BytesIO()
+                        stream.write(currBytes[i+1:])
+                        print(flush.decode('utf-8'))
+                        break
+
+            except AzureHttpError as ae:
+                if ae.status_code != 404:
+                    sys.exit(ae)
```

Comment:
What is the user experience with `sys.exit` here?

Reply:
Exit the program and print the error :) don't know if we want to use sys.exit at all for CLI ... I wrote this as a standalone python script and we copy pasted it in here pretty much. Is the standard to just print and return?

Reply:
I was thinking maybe you can just raise here and let the caller of `stream_logs` handle any exceptions? There are already a few raise in the method that the caller may have to handle anyway. Or just return if there is no error like the following one on keyboard interruption? I rarely see `sys.exit` used in CLI and IMO multiple exits isn't a good pattern anyway.